libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/index.tsx

```diff
@@ -0,0 +1,474 @@
+import { ReactNativeFile } from '@monorepo/expo/shared/apollo';
+import { CheckIcon, PlusIcon } from '@monorepo/expo/shared/icons';
+import { Colors, Radiuses, Spacings } from '@monorepo/expo/shared/static';
+import {
+  Button,
+  TextBold,
+  TextRegular,
+} from '@monorepo/expo/shared/ui-components';
+import * as React from 'react';
+import { useEffect } from 'react';
+import { Pressable, ScrollView, View } from 'react-native';
+import Modal from 'react-native-modal';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { ClientDocumentNamespaceEnum } from '../../../../apollo';
+import DriversLicense from './DriverLicense';
+import IdDocUploads from './IdDocUplods';
+import MultipleDocUploads from './MultipeDocUploads';
+import SingleDocUploads from './SingleDocUploads';
+import { Docs, ITab, IUploadModalProps } from './types';
+
+export default function UploadModal(props: IUploadModalProps) {
+  const { isModalVisible, closeModal, opacity = 0, client } = props;
+  const [tab, setTab] = React.useState<undefined | ITab>();
+  const [docs, setDocs] = React.useState<Docs>({
+    DriversLicenseFront: undefined,
+    DriversLicenseBack: undefined,
+    BirthCertificate: undefined,
+    PhotoId: undefined,
+    SocialSecurityCard: undefined,
+    ConsentForm: [],
+    HmisForm: [],
+    IncomeForm: [],
+  });
+
+  const docProps = {
+    setTab,
+    client,
+    docs,
+    setDocs,
+  };
+
+  const TABS = {
+    DriversLicense: <DriversLicense {...docProps} />,
+    BirthCertificate: (
+      <SingleDocUploads
+        docType="BirthCertificate"
+        title="Upload Birth Certificate"
+        {...docProps}
+      />
+    ),
+    PhotoId: (
+      <IdDocUploads docType="PhotoId" title="Upload Photo ID" {...docProps} />
+    ),
+    SocialSecurityCard: (
```

Comment:
why do `SocialSecurityCard` and `PhotoId` use `IdDocUploads` instead of `SingleDocUploads`?

Reply:
in the future I'm sure we will have another view of taking pictures for Ids.

Reply:
I don't think I understand. We have a screen where the user needs to upload one object (`SingleDocUploads`), and we have a screen where the user needs to upload multiple objects (`MultipleDocUploads`). Why do we need `IdDocUploads`?

Reply:
@Davit-BetterAngels pasting [this comment](https://github.com/BetterAngelsLA/monorepo/pull/519#issuecomment-2346924967) here so it's in the thread
> as IDS have different size and later we will implement when IDS camera will have outlined view to take a picture and maybe picking will have a crop we have separate component for IDS rather than singleUploads and multiple uploads

Reply:
> IDS have different size
we can pass the size as a param, though. The shared component can have a hash like.

```
THUMBNAIL_SIZES {
    "large": large_dimensions,
    "medium": medium_dimensions,
    "small": small_dimensions,
}
```
and the shared component can take a `thumbnailSize` param

> maybe picking will have a crop
maybe, but we shouldn't be duplicating components because we might do something in the future. If we need a separate component in the future, we can create it then

Reply:
done